service/dynamodb/expression/condition.go

```diff
@@ -105,3 +267,86 @@ func compareBuildCondition(c ConditionBuilder) (ExprNode, error) {
 
 	return ret, nil
 }
+
+// compoundBuildCondition is the function to make ExprNodes from And/Or
+// ConditionBuilders. There will first be checks to make sure that the input
+// ConditionBuilder has the correct format.
+func compoundBuildCondition(c ConditionBuilder) (ExprNode, error) {
+	if len(c.operandList) != 0 {
+		return ExprNode{}, fmt.Errorf("betweenBuildCondition error: unexpected number of operands, expected 0 operands")
+	}
+	if len(c.conditionList) < 2 {
+		return ExprNode{}, fmt.Errorf("betweenBuildCondition error: unexpected number of conditions, expected at least 2 conditions")
+	}
+	childNodes, err := c.buildChildNodes()
+	if err != nil {
+		return ExprNode{}, err
+	}
+	ret := ExprNode{
+		children: childNodes,
+	}
+
+	// create a string with escaped characters to substitute them with proper
+	// aliases during runtime
+	var mode string
+	switch c.Mode {
+	case AndCond:
+		mode = " AND "
+	}
+
+	ret.fmtExpr = "($c)" + strings.Repeat(mode+"($c)", len(c.conditionList)-1)
+
+	return ret, nil
+}
+
+// betweenBuildCondition is the function to make ExprNodes from Between
+// ConditionBuilders. There will first be checks to make sure that the input
+// ConditionBuilder has the correct format.
+func betweenBuildCondition(c ConditionBuilder) (ExprNode, error) {
+	if len(c.operandList) != 3 {
+		return ExprNode{}, fmt.Errorf("betweenBuildCondition error: unexpected number of operands, expected 3 operands")
+	}
+	if len(c.conditionList) != 0 {
+		return ExprNode{}, fmt.Errorf("betweenBuildCondition error: unexpected number of conditions, expected 0 conditions")
+	}
+
+	childNodes, err := c.buildChildNodes()
+	if err != nil {
+		return ExprNode{}, err
+	}
+	ret := ExprNode{
+		children: childNodes,
+	}
+
+	// Create a string with special characters that can be substituted later: $c
+	ret.fmtExpr = "$c BETWEEN $c AND $c"
+
+	return ret, nil
+}
+
+// buildChildNodes will create the list of the child ExprNodes. This avoids
+// duplication of code amongst the various buildConditions.
+func (cond ConditionBuilder) buildChildNodes() ([]ExprNode, error) {
```

Comment:
This can be simplified by removing the if and else statement, as it looks like it isn't really needed, and just always building the child nodes.